firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/vector.txt

```diff
@@ -0,0 +1,60 @@
+Scenario: The function in Util Vectors works correctly
+  GIVEN("The util resize function")
+  WHEN("the util resize function is called on 252UL buffer size")
+    THEN("Result is as expected ")
+
+  WHEN("Util function empty is used on vector")
+    THEN("Result is as expected")  
+
+  WHEN("Util function avaliable is used on vector")
+    THEN("Result is as expected")  
+
+  GIVEN("Util full function")   
+  WHEN("Util function full is used on vector")
+    THEN("Result is as expected")
+
+    
+Scenario: The function in Util push_back works correctly
+  GIVEN("the Util push_back function")
+  WHEN("the util push_back is called in bounds")
+     THEN("Result is as expected")
+
+
+  GIVEN("the Util push_back function")
+  WHEN("the util push_back is called on out of bound vector")
+     THEN("Vector is out of bound, push_back fails")
+
+
+  GIVEN("the Util push_back function ")
+  WHEN("the util push_back is called on string")
+     THEN("Result is as expceted")
+
+
+Scenario: The function in Util copy_from works correctly
+  GIVEN("the Util copy_from function")
+  WHEN("the util copy_from is called on both 4UL buffer vector")
+     THEN("Result is as expected")
+
+  WHEN("the util copy_from is called on vector of buffer 2UL and 4UL")
+    THEN("Result is as expected")
+
+  WHEN("the util copy_from is called on both same sized buffer vector")  
+    THEN("Result is as expected")
+
+
+  GIVEN("the Util copy_from function")  
+  WHEN("the Util copy_from function on array")
+     THEN("Result is as expected");
+
+  GIVEN("the Util copy_from function")
+  WHEN("the Util copy_from function on vector")
+     THEN("Result is as expected")   
+
+Scenario:  Data is erased from particular index   
+   GIVEN("the Util erase function ")
```

Comment:
The `erase` method should be tested with the following possible conditions:
- The `Vector` has more than one element before `erase`. If the `index` input parameter is in-bounds, `erase` should succeed, and the next `erase` (with an in-bounds input parameter) should also succeed. If the `index` input parameter was out-of-bounds for the first `erase` call, the `Vector` should behave as if the `erase` method hadn't been called at all.
- The `Vector` has exactly one element before `erase`. `erase(0)` should succeed (while `erase` with any other index should fail, and afterwards the `Vector` should behave as if the `erase` method hadn't been called at all), and the next `erase` should fail with a bounds error regardless of its input parameter.
- The `Vector` is empty before `erase`. `erase` should fail with a bounds error regardless of its input parameter. Afterwards, the `Vector` should behave as if the `erase` method hadn't been called at all.

In every condition, after the `erase`:

- The Vector should correctly report its size (is it empty? is it full? how much capacity is available? what is its size?).
- All indices which we expect to be out-of-bounds (e.g. for erase) should actually be out-of-bounds, while all indices which we expect to be in-bounds (e.g. for erase) should actually be in-bounds.
- All elements which were previously in the `Vector` should remain unchanged, except for the removal of the element at the specified index if the `erase` succeeded.

Reply:
The tests in the scenario for the erase method do not seem to have THENs checking whether "All elements which were previously in the Vector should remain unchanged, except for the removal of the element at the specified index if the erase succeeded.". We really do need these checks; let's address them in a future PR.